src/booking.c

```diff
@@ -1,47 +1,224 @@
         }
-        else
-        {
-            printf("Sorry only %d seats are available!\n",available_seats);
+        printf("\nPlease select one of the package: ");
+        scanf("%d", &cruise_selected);
+
+        /*
+        * Taking the count of total passengers travelling for a particular cruise selected
+        */
+        printf("\nPlease provide the total number of passengers travelling: ");
+        scanf("%d", &no_of_pass);
+
+        /*
+        * function called to check the availability for requested number of seats in the selected cruise
+        */
+        available = package_availability(cruise_selected, no_of_pass, seats_in_cruises);
+        if(available == -1) {
+            printf("Requested number of seats are available!\n");
+            booking_id = passenger_details(no_of_pass);
+            /*
+            * To Do -- call funtions suite_selection and fare_calcultion
+            */
+            do{
+                printf("\nDo you want to confirm the booking? (yes/no):  ");
+                scanf("%s",confirm_booking);
+                if( strcmp( confirm_booking, "yes") == 0 ){
+                    printf("Booking is confirmed!\n");
+                    printf("Your booking ID is: %d", booking_id);
+                    break;
+                }else{
+                    printf("\nDo you want to cancel booking? (yes/no):   ");
+                    scanf("%s",cancel_booking);
+                    if( strcmp( cancel_booking, "yes") == 0 ){
+                        printf("Booking is cancelled\n");
+                        /*
+                        * To Do need to delete all the details of that particular booking ID
+                        */
+                        break;
+                    }else{
+                       printf("\nWhat among the following changes do you want to make? \n1.Add passenger\n2.Remove passenger\n3.Room changes\n");
+                       scanf("%d",&change_option);
+                       switch(change_option){
+                            case 1: printf("\nWhat is the count of extra passengers you would like to add?");
+                                    scanf("%d",&no_of_pass);
+                                    available = package_availability(cruise_selected, no_of_pass, seats_in_cruises);
+                                    if(available == -1){
+                                        printf("Requested number of seats are available!\n");
+                                        booking_id = passenger_details(no_of_pass);
+                                        /*
+                                        * To Do -- add funtions suite_selection and fare_calcultion
+                                        */
+                                        break;
+                                    }else{
+                                         printf("Sorry:( %d seats are left\n",  available);
+                                         break;
+                                    }
```

Comment:
There is a continue condition in the flowchart which is missing in the code. Can you please update it.

Reply:
Updated file with missing condition.